gm4_auto_breeding/data/auto_breeding/functions/main.mcfunction

```diff
@@ -0,0 +1,22 @@
+#tag entities respective of their food source(s)
+tag @e[type=minecraft:pig,tag=!gm4_eats_carrots] add gm4_eats_carrots
+tag @e[type=minecraft:rabbit,tag=!gm4_eats_carrots] add gm4_eats_carrots
+tag @e[type=minecraft:pig,tag=!gm4_eats_potatoes] add gm4_eats_potatoes
+tag @e[type=minecraft:chicken,tag=!gm4_eats_seeds] add gm4_eats_seeds
+tag @e[type=minecraft:cow,tag=!gm4_eats_wheat] add gm4_eats_wheat
+tag @e[type=minecraft:llama,tag=!gm4_eats_wheat] add gm4_eats_wheat
+tag @e[type=minecraft:sheep,tag=!gm4_eats_wheat] add gm4_eats_wheat
+
+tag @e[tag=gm4_eats_carrots] add gm4_can_breed

```

Comment:
the inLove nbt tag only exists on mobs that can breed so you can check for it being 0 to tag new entities instead of using 4 tag commands. In fact, you check for inLove being 0 in your eat command so you probably don't even need this tag.